src/kubernetes.cc

```diff
@@ -101,8 +103,10 @@ class KubernetesReader::NonRetriableError
 };
 
 KubernetesReader::KubernetesReader(const Configuration& config,
+                                   const MetadataStore& store,
                                    HealthChecker* health_checker)
-    : config_(config), environment_(config), health_checker_(health_checker) {}
+    : config_(config), environment_(config), store_(store),
```

Comment:
It seems odd to me for the reader to depend on the store. Could the association-building code be moved into the updater? Possibly a task for a later refactoring.

Reply:
You're right, I think I jumped the gun on this. Ultimately, the associations for various resources would need to be looked up via the store, but for now there's only one, and it's built into the schema.